packages/loader/container-loader/closeAndGetPendingLocalState.md

```diff
@@ -0,0 +1,31 @@
+# Stashed Changes
+
+When a user goes offline, they are still able to make changes to the document. When the user goes back online, the DDSs will be asked to resubmit the ops, which will use the merging strategy of the DDS to modify the ops on top of new ops so they can be resubmitted and understood by other clients. If `Container.close()` is called while still offline, the changes will be lost. If closed while online, but the changes have not been sequenced yet (i.e. the container is "dirty"), the changes may or may not be lost.
+The stashed changes feature solves this problem by adding a new API, `Container.closeAndGetPendingLocalState()`, which will save pending changes so they may be resupplied to a new container instance and resubmitted.
+
+Additionally, the blob returned by closeAndGetPendingLocalState() contains enough data to load a container offline. This method will still return a such a blob even if the container is not dirty when called.
+
+## Using `Container.closeAndGetPendingLocalState()`
+
+When closeAndGetPendingLocalState() is called it will return a serialized blob containing the necessary data to load and resubmit the changes in a new container.
+
+The stashed changes blob can be used by passing it to `Loader.resolve()` when loading a new container. The container will then automatically submit the stashed changes when the Container reaches the "connected" state, if the changes were not originally successful.
+
+**It's important that these blobs are not reused, since it can result in the same changes being submitted multiple times, possibly resulting in document corruption.** Instead, closeAndGetPendingLocalState() should be called again on the new container, which will return a new blob containing all its pending changes, including any still-pending stashed changes it was loaded with.
+
+## How it works
```

Comment:
i like the structure from this point on. I think it's clear it's talking about implementation details and provides a concise description of the key parts of what's going on.